### ISSUE TITLE
use IAM role for authentication with Amazon EC2 Fleet Plugin

### ISSUE THREAD

**Issue by garyjyao1** [jenkinsci/ec2-fleet-plugin#176]
Hi

Similar to what's asked in https://issues.jenkins-ci.org/browse/JENKINS-55343?

Can I please ask if you can add IAM role support for this plugin?

Thanks
Gary

**Comment by terma** [comment]
hi, 

Could you please explain use case in details, thx

As far as I know assumed role credentials always has some expiration time which is relatively short [max 12h](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html)

**Comment by garyjyao1** [comment]
At my company, all IAM User keys must be rotated every 30 days. Using IAM User key and rotate every 30 days is a huge burden for Jenkins Admin (me).

If the plugin can take the IAM role of the Jenkins master, that would be great so I can get rid of using a IAM User along with key rotation.

Thanks.

**Comment by terma** [comment]
Ok make sense, here options:

- if your Jenkins master running on EC2 Instance you can assign role to that instance so Jenkins can have some IAM permissions and could assume role without any additional credentials
- else, you still need to provide kinda master credentials for Jenkins which it will use to assume role and get temporary creds

**Comment by mvantellingen** [comment]
Managing spot fleets using an IAM role always worked pretty wel. However it seems that to list the autoscaling groups credentials are required.

```console
2020-04-11 10:23:40.855+0000 [id=192] WARNING c.a.j.e.EC2FleetCloud$DescriptorImpl#doFillFleetItems: Cannot describe fleets in eu-west-1 or by endpoint
com.amazonaws.services.autoscaling.model.AmazonAutoScalingException: Request is missing Authentication Token (Service: AmazonAutoScaling; Status Code: 403; Error Code: MissingAuthenticationToken; Request ID: 1f9699d9-97e2-42b1-8ad0-2ba80e0152e2)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1799)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1383)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1359)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1139)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:796)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:764)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:738)
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:698)
at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:680)
at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:544)
at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:524)
at com.amazonaws.services.autoscaling.AmazonAutoScalingClient.doInvoke(AmazonAutoScalingClient.java:4002)
at com.amazonaws.services.autoscaling.AmazonAutoScalingClient.invoke(AmazonAutoScalingClient.java:3969)
at com.amazonaws.services.autoscaling.AmazonAutoScalingClient.invoke(AmazonAutoScalingClient.java:3958)
at com.amazonaws.services.autoscaling.AmazonAutoScalingClient.executeDescribeAutoScalingGroups(AmazonAutoScalingClient.java:1556)
at com.amazonaws.services.autoscaling.AmazonAutoScalingClient.describeAutoScalingGroups(AmazonAutoScalingClient.java:1527)
at com.amazon.jenkins.ec2fleet.fleet.AutoScalingGroupFleet.describe(AutoScalingGroupFleet.java:41)
at com.amazon.jenkins.ec2fleet.EC2FleetCloud$DescriptorImpl.doFillFleetItems(EC2FleetCloud.java:791)
```

**Comment by kandia** [comment]
Hello,

Could you please merge @vroy's pull request ? 

I Have the same issues. with an IAM User i can see both spot fleets and autoscaling group in the plugin configuration. But when i use instance profile i can see only spot fleets and I've got the same error decribed above by @mvantellingen .

Thx for you help.

**Comment by kandia** [comment]
I've just tested the pull request of @vroy and now I can see both spot fleet request and autoscaling group available using the instance profile.

hope that this work will be included in next release. 

Thx

**Comment by vroy** [comment]
I believe that this can now be closed now that #190 has been merged.
